### PR TITLE
Artifact improvements and artifact debug QOL

### PR DESCRIPTION
**Pull request by MylieDaniels**
#### Summary
Features "Artifact improvements and artifact debug QOL"

#### Purpose of change
Artifacts could not be configured to provide effects except when wielded if not armor, or worn if armor.
In addition, artifact debugging was not at all configurable, and always spawned an alien_reality artifact with 5 max attributes, 1000 power, and -2000 max negative power.

#### Describe the solution
- Artifacts can now have enchantment::has (aka where they apply from, be it while worn, while wielded, or even while carried in a bag) specified for each individual enchantment entry- except for the activated effect, which logically remains wield-only as you can only activate it while wielding the artifact.
- This does work for hit_me and hit_you effects, for any amazing things that can be done with an object that casts magic from your backpack when you punch something.
- map::spawn_artifact now supports specification of max_attributes, power_level, and max_negative_power.
- Added example (but hopefully actually usable) artifact procgen data for artifacts that are still effective while held in a bag or pocket or thermos or whatever
- Debug spawning of artifacts vastly expanded, now supports selection of group, max attributes, power level, and max negative power for a fully customizable artifact testing experience!
- Added new material for artifacts, Monolith, which exists to provide easier support for artifacts being quite sturdy, and added two artifacts made from it that tie to the new example artifact category
- In addition, one misleading error message was reworded, and artifacts no longer have a niche case that causes num_attribute to increase without actually adding an attribute.
- Updated documentation on artifacts to include new parameter

#### Describe alternatives you've considered
Split this into two PR's, one to add the capability and one to add the example items.

#### Testing
Spawned in numerous artifacts, lugged them around to ensure they gave stats, threw/dropped/deleted them to ensure the effects went away, ran them over with a pickup truck, ran them over with a train, tested various explosives on them- the train and a military barrel bomb both managed to damage the Monolith items.

### CODE REVIEW

File: src/map.cpp [hunk 1]
```diff
@@ -4342,12 +4342,13 @@ std::vector<item *> map::spawn_items( const tripoint &p, const std::vector<item>
     return ret;
 }
 
-void map::spawn_artifact( const tripoint &p, const relic_procgen_id &id )
+void map::spawn_artifact( const tripoint &p, const relic_procgen_id &id, const int &max_attributes = 5,
+                          const int &power_level = 1000, const int &max_negative_power = -2000 )
```
Comment (BrettDong) [inline src/map.cpp:4346]:
```suggestion
void map::spawn_artifact( const tripoint &p, const relic_procgen_id &id, const int max_attributes = 5,
                          const int power_level = 1000, const int max_negative_power = -2000 )
```

File: src/map.h [hunk 2]
```diff
@@ -1210,7 +1210,8 @@ class map
         void i_rem( const point &p, item *it ) {
             i_rem( tripoint( p, abs_sub.z ), it );
         }
-        void spawn_artifact( const tripoint &p, const relic_procgen_id &id );
+        void spawn_artifact( const tripoint &p, const relic_procgen_id &id, const int &max_attributes = 5,
+                         const int &power_level = 1000, const int &max_negative_power = -2000 );
```
Comment (BrettDong) [inline src/map.h:1214]:
```suggestion
        void spawn_artifact( const tripoint &p, const relic_procgen_id &id, const int max_attributes = 5,
                         const int power_level = 1000, const int max_negative_power = -2000 );
```

File: src/debug_menu.cpp [hunk 3]
```diff
@@ -2470,11 +2469,23 @@ void debug()
...
-                artifact_natural_property prop = static_cast<artifact_natural_property>( rng( ARTPROP_NULL + 1,
-                                                 ARTPROP_MAX - 1 ) );
-                here.create_anomaly( *center, prop );
-                here.spawn_artifact( *center, relic_procgen_data_alien_reality );
+            uilist relic_menu;
+            std::vector<T> relic_list = relic_procgen_data::get_all()
+            relic_menu.text = _( "Choose artifact data:" );
+            int menu_ind = 0;
+            for( auto &elem : relic_list ) {
+                relic_menu.addentry( menu_ind, true, MENU_AUTOASSIGN, _( "%1$s" ), elem );
```
Comment (BrettDong) [inline src/debug_menu.cpp:2477]:
```suggestion
                relic_menu.addentry( menu_ind, true, MENU_AUTOASSIGN, elem );
```

File: src/relic.cpp [hunk 4]
```diff
@@ -761,3 +771,8 @@ bool operator==( const relic &source_relic, const relic &target_relic )
     }
     return is_the_same;
 }
+
+const std::vector<relic_procgen_data> &relic_procgen_data::get_all()
+{
+    return relic_procgen_data_factory.get_all();
+}
```
Comment (Maleclypse) [inline src/relic.cpp:778]: missing a newline at end of file according to the tests.

File: data/json/materials.json [hunk 5]
```diff
@@ -1752,6 +1752,26 @@
     "cut_dmg_verb": "chipped",
     "burn_products": [ [ "pebble", 1 ] ]
   },
+  {
+    "type": "material",
+    "id": "monolith",
```
Comment (Maleclypse) [inline data/json/materials.json:1757]: Are you wanting all the artifacts to use this material or just random artifacts, or even only some random artifacts?

Comment (Mikenyes) [reply data/json/materials.json:1757]: I would like to convert the other artifacts to this material, and will if it's alright with the devs.

File: src/relic.cpp (hunk 4 above)
Comment (Mikenyes) [reply src/relic.cpp]: Fixed, thank you.

**Pull request merged**